make/docker.mk

```diff
@@ -0,0 +1,85 @@
+ifndef DOCKER_MK
+DOCKER_MK:=# Prevent repeated "-include".
+
+# If running in Jenkins we don't allow for interactively running the container
+DOCKER_RUN_INTERACTIVE_SWITCH = -i
+ifneq ($(BUILD_TAG),)
+	DOCKER_RUN_INTERACTIVE_SWITCH =
+endif
+
+.PHONY: docker-image-deploy
+## Build the docker image that can be deployed (only contains bare operator)
+docker-image-deploy: Dockerfile
```

Comment:
IMHO we should use ` operator-sdk build ` to build the image

Reply:
I was told to model the devconsole-operator by @nikhil-thomas do we want to change this?

Reply:
I am with @sthaha (especially as, some of us don't even have `docker` installed :sweat_smile: )

Reply:
Ok I will use operator-sdk as much as possible

Reply:
the only reason not to use `operator-sdk build` is a need for supplying go build flags.
it is a known issue https://github.com/operator-framework/operator-sdk/issues/1435

in devconsole operator they want to tag their binary with commit-hash build time
here https://github.com/operator-framework/operator-sdk/issues/1435

as we are not doing it right now, we can use `operator-sdk build`